### PR TITLE
API - First parent fix and ancestors addition

### PR DESCRIPTION
**Pull request by dwsutherland**
These changes partially address https://github.com/cylc/cylc-ui/issues/229#

None first parent:
```
    [[bar]]
        inherit = None, FAM3
```
Now set to root `firstParent`:
```
          {
            "id": "sutherlander|baz|20170101T0000+13|bar",
            "state": "waiting",
            "cyclePoint": "20170101T0000+13",
            "latestMessage": "",
            "firstParent": {
              "id": "sutherlander|baz|20170101T0000+13|root",
              "state": "waiting"
            },
            "parents": [
              {
                "id": "sutherlander|baz|20170101T0000+13|FAM3"
              }
            ],
            "ancestors": [
              {
                "id": "sutherlander|baz|20170101T0000+13|root"
              }
            ],
            "namespace": [
              "root",
              "FAM",
              "FAM2",
              "FAM3",
              "bar"
            ],
            "task": {
              "meanElapsedTime": 0,
              "name": "bar"
            },
            "jobs": []
```
First parent ancestors now added to `taskProxies` and `familyProxies`:
```
      ancestors (sort: {keys: ["depth"]}) {
        id
      }
```
```
        "taskProxies": [
          {
            "id": "sutherlander|baz|20170101T0000+13|foo",
            "state": "waiting",
            "cyclePoint": "20170101T0000+13",
            "latestMessage": "",
            "firstParent": {
              "id": "sutherlander|baz|20170101T0000+13|FAM3",
              "state": "waiting"
            },
            "parents": [
              {
                "id": "sutherlander|baz|20170101T0000+13|FAM3"
              }
            ],
            "ancestors": [
              {
                "id": "sutherlander|baz|20170101T0000+13|root"
              },
              {
                "id": "sutherlander|baz|20170101T0000+13|FAM"
              },
              {
                "id": "sutherlander|baz|20170101T0000+13|FAM2"
              },
              {
                "id": "sutherlander|baz|20170101T0000+13|FAM3"
              }
            ],
            "namespace": [
              "root",
              "FAM",
              "FAM2",
              "FAM3",
              "foo"
            ],
            "task": {
              "meanElapsedTime": 0,
              "name": "foo"
            },
            "jobs": []
          },
```

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Already covered by existing tests.
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/ws_data_mgr.py [hunk 1]
```diff
@@ -356,18 +358,16 @@ def generate_ghost_families(self, family_proxies=None, cycle_points=None):
             # first-parent single-inheritance tree
             if not cycle_points or point_string not in cycle_points:
                 continue
-            cycle_first_parents = set()
+            cycle_first_parents = []
```
Comment (kinow) [inline cylc/flow/ws_data_mgr.py]: Are we going to have duplicate first parents? Is that why we are replacing the list by a set now?

Comment (dwsutherland) [reply cylc/flow/ws_data_mgr.py]: Hmm... Actually yes, possibly, I wanted to preserve order.. Hold on will change it

Comment (dwsutherland) [reply cylc/flow/ws_data_mgr.py]: I don't think order actually matters for this XD.. my bad.

**Comment by kinow** [comment]
Tested again after last commit/change, everything working fine with no issues. 👍 thanks @dwsutherland, re-approved!

**Comment by dwsutherland** [comment]
Two approvals.. merged.

**Pull request merged**